test/e2e/service_export_test.go

```diff
@@ -55,11 +55,16 @@ func TestServiceExport(t *testing.T) {
 	t.Log("create service with byo revision")
 	serviceCreateWithOptions(r, "hello", "--revision-name", "rev1")
 
+	userImage := pkgtest.ImagePath("helloworld")
+	if strings.Contains(userImage, "@") {
+		// Images specified by digest are not set in user-image annotation.
+		userImage = ""
+	}
 	t.Log("export service-revision1 and compare")
 	serviceExport(r, "hello", test.BuildServiceWithOptions("hello",
 		servingtest.WithConfigSpec(test.BuildConfigurationSpec()),
 		servingtest.WithBYORevisionName("hello-rev1"),
-		test.WithRevisionAnnotations(map[string]string{"client.knative.dev/user-image": pkgtest.ImagePath("helloworld")}),
+		test.WithRevisionAnnotations(map[string]string{"client.knative.dev/user-image": userImage}),
```

Comment:
Shouldn't the whole annotation not be set if the userImage is "" (instead of setting the annotation with an empty string ?)

Reply:
I've chosen this approach to keep the tests simple. In the end the annotation is set to an empty value by the kn client anyway: https://github.com/knative/client/blob/release-0.23/pkg/kn/commands/service/create.go#L261

Reply:
but as it seems to work upstream also with an empty value, looks fine to me. Please note though that this fix might not work with `--lock-to-digest` option as this option relies on this annotation to contain the value with digest (and if I think twice, I think this annotation is supposed to always hold the digest, but let me check)

Reply:
Ah, no. it's the other way round, actually the annotation shouldn't contain any digest, so the fix looks good to me.

/approve
/lgtm

Reply:
Thanks for checking!